### PR TITLE
Flowers dataset added

### PR DESCRIPTION
**Pull request by alecokas**
* Full functionality except for cropping
* There is a reasonable amount of repeated code in the dataset code but I really don't want to make the inheritance structure even more complicated at this stage

### CODE REVIEW

File: trainers/image_trainer.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ def __init__(self, model, batch_size, save_location,
                 results from training the model.
         """
         super().__init__(model, batch_size, save_location, show_progress_bar)
+        self.noise_size = kwargs.get('conditional_emb_size')
```
Comment (Devin-Taylor) [inline trainers/image_trainer.py:22]: this is wrong, noise_size and conditional_emb_size are different parameters. One is 100 and the other is 128

File: utils/data_helpers.py [hunk 2]
```diff
@@ -204,26 +211,60 @@ def download_cub(include_text=False):
...
+            with zipfile.ZipFile(flowers_text_backup_location, 'r') as zipfd:
+                zipfd.extractall('data/')
+        else:
+            # Clean download
+            print('Downloading flowers text from Google Drive ID: {}'.format(FLOWERS_TEXT_GDRIVE_ID))
+            gdd.download_file_from_google_drive(file_id=FLOWERS_TEXT_GDRIVE_ID,
+                                                dest_path=flowers_text_download_location,
+                                                unzip=True)
+            mkdir('data/backup')
+            shutil.copy(flowers_text_download_location, flowers_text_backup_location)
```
Comment (Devin-Taylor) [inline utils/data_helpers.py:259]: we should definitely make this backup functionality a function but not sure how straight forward that is, might be too much custom code per use case

**Comment by Devin-Taylor** [comment]
Looks good, just the dimensionality fix then happy to merge

File: utils/data_helpers.py (hunk 2 above)
Comment (alecokas) [reply utils/data_helpers.py]: Done :+1:

File: trainers/image_trainer.py (hunk 1 above)
Comment (alecokas) [reply trainers/image_trainer.py]: Very true - thanks

**Pull request merged**